File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        s=str(x)
        lp=0
        rp=len(s)-1
        while (lp<=rp):
            if (s[lp]) == (s[rp]):
                lp+=1
                rp-=1 
            else:
                return False
        return True           

File: 0009-palindrome-number/0009-palindrome-number.java
import java.util.*;
class Solution {
    public boolean isPalindrome(int x) {
        String a;
        String b;
        String rev="";
        if(x>=0)
        {
            a=String.valueOf(x);
            for(int i=a.length()-1;i>=0;i--)
            {
                rev=rev+a.charAt(i);

            }
        }
        else
        {
            x=x*(-1);
            a=String.valueOf(x);
            a="-"+a;
            for(int i=a.length()-1;i>=0;i--)
            {
                rev="-"+rev+a.charAt(i);

            }
        }
        
        if(a.equals(rev))
        {
            return true;
        }
        else
        {
            return false;
        }
        
    }
}